cpp/src/arrow/filesystem/s3_internal.h

```diff
@@ -154,8 +154,9 @@ Status ErrorToStatus(const std::string& prefix, const std::string& operation,
   // https://sdk.amazonaws.com/cpp/api/LATEST/namespace_aws_1_1_s3.html#ae3f82f8132b619b6e91c88a9f1bde371
   return Status::IOError(
       prefix, "AWS Error ",
-      S3ErrorToString(static_cast<Aws::S3::S3Errors>(error.GetErrorType())), " during ",
-      operation, " operation: ", error.GetMessage());
+      S3ErrorToString(static_cast<Aws::S3::S3Errors>(error.GetErrorType())),
+      " (http status ", static_cast<int>(error.GetResponseCode()), ")",
```

Comment:
Shouldn't we add this only if the AWS error is actually "unknown"?

Reply:
Yes you are right, that will be less confusing I think (instead of the user needing to juggle with two errors).